File: Basic Math/Reverse digits/ReverseDigits.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
 public:
  long long int reverse_digit(long long int n) {
    long long int res = 0, rem = 0;
    while (n > 0) {
      rem = n % 10;
      res = res * 10 + rem;
      n /= 10;
    }
    return res;
  }
};

int main() {
  int T;
  cin >> T;
  while (T--) {
    long long int n;
    cin >> n;
    Solution ob;
    long long int ans = ob.reverse_digit(n);
    cout << ans << "\n";

    cout << "~" << "\n";
  }
  return 0;
}

File: Basic Math/Reverse digits/ReverseDigits.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class Solution {
  public long reverse_digit(long n) {
    long res = 0, rem = 0;
    while (n > 0) {
      rem = n % 10;
      res = res * 10 + rem;
      n /= 10;
    }
    return res;
  }
}

class ReverseDigits {
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    int T = Integer.parseInt(br.readLine().trim());
    while (T-- > 0) {
      long n = Long.parseLong(br.readLine().trim());
      Solution ob = new Solution();
      long ans = ob.reverse_digit(n);
      System.out.println(ans);

      System.out.println("~");
    }
  }
}
